packages/loader/container-loader/src/test/deltaManager.spec.ts

```diff
@@ -313,6 +332,95 @@ describe("Loader", () => {
                     // make extra sure
                     await tickClock(expectedTimeout);
                 });
+
+                it("Should throw error with gap in client seq num", async () => {
+                    await startDeltaManager();
+
+                    deltaManager.inbound.on("error",(error) => { expectedError = error; } );
+
+                    await emit(1, MessageType.Operation);
+
+                    // send op with gap in clientSeqNum
+                    deltaConnection.emitOp(docId, [{
+                        clientId: "test",
+                        clientSequenceNumber: clientSeqNumber + 2,
+                        minimumSequenceNumber: 0,
+                        sequenceNumber: seq++,
+                        type: MessageType.Operation,
+                    } as any as ISequencedDocumentMessage]);
+
+                    await yieldEventLoop();
+                    assert.strictEqual(expectedError.message, "gap in client sequence number: 1")
+                });
+
+                it("Should pass with one noop sent, 0 received and one gap", async () => {
+                    await startDeltaManager();
+
+                    deltaManager.inbound.on("error",(error) => { expectedError = error; } );
+
+                    await emit(1, MessageType.Operation);
+
+                    // send 1 noop without receiving
+                    deltaManager.submit(MessageType.NoOp);
+
+                    // send op with gap in clientSeqNum
+                    deltaManager.submit(MessageType.Operation);
+                    deltaConnection.emitOp(docId, [{
+                        clientId: "test",
+                        clientSequenceNumber: clientSeqNumber+2,
+                        minimumSequenceNumber: 0,
+                        sequenceNumber: seq++,
+                        type: MessageType.Operation,
+                    } as any as ISequencedDocumentMessage]);
+
+                    await yieldEventLoop();
```

Comment:
I think yieldEventLoop() is sufficient, but you might also assert here that DetalManager actually processed all the ops generated by this test case (i.e. something like assert(deltaManager.lastMessage.sequenceNumber === seq)); If yieldEventLoop() does not wait for all promises to settle for some reason, then this test could pass (if assert is not added) even if it would fail later on. I.e. adding assert ensures that test actually went through all ops mentioned here (op processing is async).